spec/controllers/internal_api/v1/pillars_controller_spec.rb

```diff
@@ -434,97 +434,107 @@
     end
   end
 
-  def expected_dex_json(num, certificate)
-    {
-      id:              num,
-      name:            "LDAP Server #{num}",
-      root_ca_data:    Base64.encode64(certificate.certificate),
-      bind:            {
-        anonymous: false,
-        dn:        "cn=admin,dc=ldap_host_#{num},dc=com",
-        pw:        "pass"
-      },
-      username_prompt: "Username",
-      user:            {
-        base_dn:  "cn=users,dc=ldap_host_#{num},dc=com",
-        filter:   "(objectClass=person)",
-        attr_map: {
-          username: "uid",
-          id:       "uid",
-          email:    "mail",
-          name:     "name"
-        }
-      },
-      group:           {
-        base_dn:  "cn=groups,dc=ldap_host_#{num},dc=com",
-        filter:   "(objectClass=group)",
-        attr_map: {
-          user:  "uid",
-          group: "member",
-          name:  "name"
-        }
-      }
-    }
-  end
-
-  # rubocop:disable RSpec/ExampleLength
-  context "with dex LDAP connectors tls" do
-    it "has dex LDAP connectors" do
-      dex_connector_ldap = create(:dex_connector_ldap, :tls, :regular_admin)
-      CertificateService.create(service: dex_connector_ldap, certificate: certificate)
+  context "with Dex LDAP connectors" do
+    before do |example|
+      # calculate a likely-unique 0-100 integer hash value by adding up the
+      # integer values of chars in the example name (context + example)
+      u = example.metadata[:full_description].codepoints.reduce(0, :+) % 100
+      @conn_name = "LDAP connector #{u}"
+      @host_name = "ldap_host_#{u}.com"
+    end
 
-      expected_json = {
-        registries:          [],
-        kubelet:             {
-          :"compute-resources" => {},
-          :"eviction-hard"     => ""
+    def expected_dex_json(num, certificate, c_name, host)
+      {
+        type:            "ldap",
+        id:              num,
+        name:            c_name,
+        root_ca_data:    Base64.encode64(certificate.certificate),
+        bind:            {
+          anonymous: false,
+          dn:        "cn=admin,dc=#{host.chomp(".com")},dc=com",
```

Comment:
If the already passed in 'num' matches the id, as it is still constructing a json object checking that, then simply using number and constructing the hostname in place is much cleaner. So long as 'num' is available there to use, I fail to see how this is an improvement to the expected_dex_json function.